sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/implementation/Utils.java

```diff
@@ -0,0 +1,161 @@
+        if (matcher.noMatch()) {
+            throw new IllegalArgumentException(String.format("Illegal memory format : %s", memory));
+        }
+        if (matcher.matchGB()) {
+            return Double.valueOf(matcher.getGBString());
+        } else { // Mb
+            return BigDecimal.valueOf(Long.parseLong(matcher.getMBString()))
+                .divide(D_1024, 1, RoundingMode.CEILING)
+                .doubleValue();
+        }
+    }
+
+    /**
+     * Converts cpu count from double to String
+     * @param cpuCount cpu count, 1 core can be represented by 1 or 1000m
+     * @return cpu String
+     */
+    public static String toCpuString(double cpuCount) {
+        BigDecimal cpuDecimal = BigDecimal.valueOf(cpuCount);
+        if (isInteger(cpuCount)) {
+            return String.valueOf(cpuDecimal.intValue());
+        } else {
+            return String.format("%dm", cpuDecimal.multiply(D_1000).intValue());
+        }
+    }
+
+    /**
+     * Converts memory count from double to String
+     * @param sizeInGB memory in GB
+     * @return memory String
+     */
+    public static String toMemoryString(double sizeInGB) {
+        BigDecimal memoryDecimal = BigDecimal.valueOf(sizeInGB);
+        if (isInteger(sizeInGB)) {
+            return String.format("%dGi", memoryDecimal.intValue());
+        } else {
+            return String.format("%dMi", memoryDecimal.multiply(D_1024).intValue());
+        }
+    }
+
+    private static boolean isInteger(double sizeInGB) {
+        return (sizeInGB % 1) == 0;
+    }
+
+    private static class CpuMatcher {
+        static final Pattern CPU_INTEGER = Pattern.compile("^[0-9]+$");     // 1, 2, 3
+        static final Pattern CPU_FRACTION = Pattern.compile("^([0-9]+)m$"); // 500m, 1000m
+        Matcher integerMatcher;
+        Matcher fractionMatcher;
+
+        CpuMatcher(Matcher integerMatcher, Matcher fractionMatcher) {
+            this.integerMatcher = integerMatcher;
+            this.fractionMatcher = fractionMatcher;
+        }
+
+        static CpuMatcher create(String cpu) {
+            Matcher integerMatcher = CPU_INTEGER.matcher(cpu);
+            Matcher fractionMatcher = CPU_FRACTION.matcher(cpu);
+            return new CpuMatcher(
+                integerMatcher.matches() ? integerMatcher : null,
+                fractionMatcher.matches() ? fractionMatcher : null
+            );
+        }
+
+        boolean noMatch() {
+            return integerMatcher == null && fractionMatcher == null;
+        }
+
+        boolean matchFraction() {
+            return this.fractionMatcher != null;
+        }
+
+        String getFraction() {
+            return this.fractionMatcher.group(1);
+        }
+    }
+
+    private static class MemoryMatcher {
+        static final Pattern MEMORY_GB = Pattern.compile("^([0-9]+)Gi"); // 1Gi, 2Gi
+        static final Pattern MEMORY_MB = Pattern.compile("^([0-9]+)Mi"); // 512Mi, 1024Mi
```

Comment:
Try double check with service that they will honor case (e.g. they won't change return value from 512Mi to 512mi or 512MI).

Reply:
From the regex they give:
@"^(?<number>0|-?[1-9][0-9]*)(?<unit>Gi|Mi)$ for mem 
@"^(?<number>0|-?[1-9][0-9]*)(?<unit>m?)$ for cpu

It is case-sensitive.